tests/DataObject/DataObjectDocumentTest.php

```diff
@@ -85,6 +89,57 @@ public function testShouldIndex(): void
         $this->assertFalse($doc->shouldIndex());
     }
 
+    public function testShouldIndexChild(): void
+    {
+        $config = $this->mockConfig();
+
+        $parent = $this->objFromFixture(Page::class, 'page1');
+        // Make sure our Parent is published before we fetch our child pages
+        $parent->publishRecursive();
+
+        $childOne = $this->objFromFixture(Page::class, 'page2');
+        $childTwo = $this->objFromFixture(Page::class, 'page3');
+
+        // Publish childOne
+        $childOne->publishRecursive();
+        // Need to re-fetch childOne so that our Versioned state is up-to-date with what we just published
+        $childOne = $this->objFromFixture(Page::class, 'page2');
+
+        // $docOne has a published page
+        $docOne = DataObjectDocument::create($childOne);
+        // $docTwo has an unpublished page
+        $docTwo = DataObjectDocument::create($childTwo);
+
+        // Add both documents to our indexes, as this isn't the functionality we're testing here
+        $config->set(
+            'getIndexesForDocument',
+            [
+                $docOne->getIdentifier() => [
+                    'index' => 'data',
+                ],
+                $docTwo->getIdentifier() => [
+                    'index' => 'data',
+                ],
+            ]
+        );
+
+        // Our parent page has been published, and so has our child page, so this should be indexable
+        $this->assertTrue($docOne->shouldIndex());
+        // Our parent page has been published, but the child has not
+        $this->assertFalse($docTwo->shouldIndex());
+
+        // Now trigger a change on our parent page (so that the draft and live versions no longer match)
```

Comment:
Would it be worth covering a scenario where a change to a child page is published downstream of the parent's draft changes?

Reply:
Added in https://github.com/silverstripe/silverstripe-search-service/pull/77/commits/039bef90ddef1bab5dac8517f53a9aaa1ca5e98e 😃